common/inventory_parser/inventory_parser.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/inventory_parser.hpp>
+
+#include <boost/filesystem.hpp>
+#include <common/key_value_parser.hpp>
+#include <common/processes.hpp>
+#include <common/log.hpp>
+
+namespace mender {
+namespace common {
+namespace inventory_parser {
+
+using namespace std;
+namespace kvp = mender::common::key_value_parser;
+namespace procs = mender::common::processes;
+namespace log = mender::common::log;
+namespace err = mender::common::error;
+namespace fs = boost::filesystem;
+
+kvp::ExpectedKeyValuesMap GetInventoryData(const string generators_dir) {
+	bool any_success;
+	bool any_failure;
+	kvp::KeyValuesMap data;
+
+	fs::path dir_path(generators_dir);
+	if (!fs::exists(dir_path)) {
+		return kvp::ExpectedKeyValuesMap(data);
+	}
+
+	fs::directory_iterator end_itr; // default construction yields past-the-end
```

Comment:
Come again.... :persevere:

Reply:
? This how `boost::filesystem` does it, unfortunately.

Reply:
Buu @boost

Reply:
Cannot agree more!